libs/native/src/main21/java/org/elasticsearch/nativeaccess/jdk/JdkVectorLibrary.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.nativeaccess.jdk;
+
+import org.elasticsearch.nativeaccess.lib.VectorLibrary;
+
+import java.lang.foreign.FunctionDescriptor;
+import java.lang.foreign.MemorySegment;
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
+
+import static java.lang.foreign.ValueLayout.ADDRESS;
+import static java.lang.foreign.ValueLayout.JAVA_BYTE;
+import static java.lang.foreign.ValueLayout.JAVA_INT;
+import static org.elasticsearch.nativeaccess.jdk.LinkerHelper.downcallHandle;
+
+public final class JdkVectorLibrary implements VectorLibrary {
+
+    static {
+        System.loadLibrary("vec");
+    }
+
+    public JdkVectorLibrary() {}
+
+    static final MethodHandle dot8stride$mh = downcallHandle("dot8s_stride", FunctionDescriptor.of(JAVA_INT));
+    static final MethodHandle sqr8stride$mh = downcallHandle("sqr8s_stride", FunctionDescriptor.of(JAVA_INT));
+
+    static final MethodHandle dot8s$mh = downcallHandle("dot8s", FunctionDescriptor.of(JAVA_INT, ADDRESS, ADDRESS, JAVA_INT));
+    static final MethodHandle sqr8s$mh = downcallHandle("sqr8s", FunctionDescriptor.of(JAVA_INT, ADDRESS, ADDRESS, JAVA_INT));
+
+    // Stride of the native implementation - consumes this number of bytes per loop invocation.
+    // There must be at least this number of bytes/elements available when going native
+    static final int DOT_STRIDE = 32;
+    static final int SQR_STRIDE = 16;
```

Comment:
I still think this could be a problem with x64, where strides are likely to be longer - having no const here, but just initialize with sqr8Stride()/dot8Stride() will give more benefit.
But we'll cross that bridge when we'll come to it. Looks OK for now.